finder-app/writer.c

```diff
@@ -0,0 +1,83 @@
+/***********************************************************************
+* @file  writer.c
+* @version 0
+* @brief  Implementation of writer file
+*
+* @author Iyona Lynn Noronha, iyonalynn.noronha@Colorado.edu
+*
+* @institution University of Colorado Boulder (UCB)
+* @course   ECEN 5713 - Advanced Embedded Software Development
+* @instructor Dan Walkes
+*
+* Revision history:
+*   0 Initial release.
+*
+* @resources
+* 1. Command Line Arguments: https://www.geeksforgeeks.org/command-line-arguments-in-c-cpp/
+* 2. Check directory is there or not: https://stackoverflow.com/questions/12510874/how-can-i-check-if-a-directory-exists
+* 3. Chatgpt promt: Extract directory path and check if it exists
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <syslog.h>
+#include <errno.h>
+#include <sys/stat.h>
+
+int main(int argc, char *argv[]) {
+
+    // Error logging start
+    openlog("writer", LOG_PID, LOG_USER);
+	
+    if (argc != 3) {
+		syslog(LOG_ERR, "Error: Two arguments required");
+        printf("Error: Two arguments required\n");
+        printf("Usage: ./writer <writefile> <writestr>\n");
+        return 1;
+    }
+
+    const char *writefile = argv[1];
+    const char *writestr = argv[2];
+	
+    // Extract directory path
+    char *dir = strdup(writefile);
+    char *slash = strrchr(dir, '/'); // Find last '/'
```

Comment:
This is going to find the first `/` I believe.  There may be a problem here if you used an absolute path like `/home/some/file/blah`, I think you'll pass an empty string to `stat` below.